dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/LibertyHibernateValidatorExtension.java

```diff
@@ -109,8 +110,8 @@ private ValidationExtension delegate(String newClassloaderHint) {
                         Tr.debug(tc, "Set Class loader back to " + oldClassLoader);
                     }
                 }
-                if (setClassLoader != null && setClassLoader.wasChanged) {
-                    releaseLoader(tcclClassLoader);
+                if (tuple != null && tuple.wasCreatedViaClassLoadingService) {
```

Comment:
To make sure I understand the nature of this fix.  I think the problem could only occur `if (newClassloaderHint !=null)` because in that case the code above would wrapper the class loader in a `Validation20ClassLoader` instance and that would always force a replacement of the TCCL even if the expected TCCL instance was already set.  So that would force us to release the existing (previous) TCCL even though we may not have created it.  Is my understanding correct?

Reply:
Yes, for sure that case, but it's also possible that the TCCL is inadvertently destroyed (without being created) in a case where the current thread's context classloader is set to anything other than `AppClassLoader` or `ThreadContextClassLoader` - so I think it could occur when the thread's context classloader is set to an OSGi loader like "ContextFinder", etc.